### PR TITLE
RSE-1448: Unit Tests For Applicant Management Post Processing Classes

### PR DESCRIPTION
**Pull request by shahrukh-compucorp**
## Overview
This pr adds tests for classes that are responsible for applicant management custom group post processing.

## Technical Details
This pr adds tests for following classes

- CRM_CiviAwards_Service_ApplicantManagementAwardDetailPostProcessor.
- CRM_CiviAwards_Service_ApplicantManagementCustomGroupPostProcessor.
- CRM_CiviAwards_Service_ApplicantManagementCustomGroupDisplayFormatter.

### CODE REVIEW

File: tests/phpunit/CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatterTest.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   * Test process display function.
+   */
+  public function testProcessDisplay() {
+    $caseCategories = [3 => 'Awards'];
+    $expectedDisplay = CaseCategoryForCustomGroupSupport::AWARDS_CATEGORY_CG_LABEL;
+    $cgExtends = ['Awards' => $expectedDisplay];
+    $subTypes = [3 => [1 => 1, 2 => 2]];
+    $awardManagementHelper = $this->getApplicantManagementHelperMock($caseCategories, $cgExtends, $subTypes);
+    $row = ['id' => 3, 'extends_entity_column_id' => 3];
+    $displayFormatter = $this->getMockBuilder(ApplicantManagementCustomGroupDisplayFormatter::class)
```
Comment (tunbola) [inline tests/phpunit/CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatterTest.php:24]: Can we move this to a new method? `getDsiplayFormatter`

**Review by tunbola — CHANGES_REQUESTED** [review]

File: CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatter.php [hunk 2]
```diff
@@ -63,9 +63,19 @@ public function processDisplay(array &$row) {
...
     $row['extends_entity_column_value'] = $relevantSubTypes;
   }
 
+  /**
+   * Provide the list of sub types.
+   *
+   * @return array
+   *   List of sub types.
+   */
+  public function buildOptions() {
```
[the remark this answers is not in the document]
Comment (shahrukh-compucorp) [reply CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatter.php]: fixed

File: tests/phpunit/CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatterTest.php (hunk 1 above)
Comment (shahrukh-compucorp) [reply tests/phpunit/CRM/CiviAwards/Service/ApplicantManagementCustomGroupDisplayFormatterTest.php]: fixed

**Review by tunbola — APPROVED** [review]

File: CRM/CiviAwards/Helper/CustomGroupPostProcess.php [hunk 3]
```diff
@@ -131,4 +132,14 @@ public function getCustomGroupSubTypesList($customGroupId = NULL) {
     return !empty($customGroupSubTypes[$customGroupId]) ? $customGroupSubTypes[$customGroupId] : [];
   }
 
+  /**
+   * Provide the list of sub types.
+   *
+   * @return array
+   *   List of sub types.
+   */
+  public function buildOptions() {
```
Comment (tunbola) [inline CRM/CiviAwards/Helper/CustomGroupPostProcess.php:141]: Can we rename to getAwardSubTypes ?

**Review by tunbola — CHANGES_REQUESTED** [review]
.

**Review by tunbola — APPROVED** [review]

**Pull request merged**